include/boost/lexical_cast/bad_lexical_cast.hpp

```diff
@@ -60,23 +60,29 @@ namespace boost
         {}
 
 #ifndef BOOST_NO_TYPEID
+#ifdef BOOST_NO_STD_TYPEINFO
+# define BOOST_LEXICAL_CAST_TYPE_INFO ::type_info
```

Comment:
How about a
```
namespace detail {
struct type_info_workaround { 
#ifdef BOOST_NO_STD_TYPEINFO
    typedef ::type_info type;
#else
    typedef ::std::type_info type; 
};
}
```
There's less macro in that way

Reply:
Agree, I have updated the code, please take a look.